azure-kusto-data/azure/kusto/data/data_format.py

```diff
@@ -2,22 +2,37 @@
 # Licensed under the MIT License
 from enum import Enum
 
+class IngestionMappingType(Enum):
+    CSV = "Csv"
+    JSON = "Json"
+    AVRO = "Avro"
+    PARQUET = "Parquet"
+    ORC = "Orc"
+    W3CLOGFILE = "W3CLogFile"
+    UNKNOWN = "Unknown"
+
 
 class DataFormat(Enum):
     """All data formats supported by Kusto."""
 
-    CSV = "csv"
-    TSV = "tsv"
-    SCSV = "scsv"
-    SOHSV = "sohsv"
-    PSV = "psv"
-    TXT = "txt"
-    JSON = "json"
-    SINGLEJSON = "singlejson"
-    AVRO = "avro"
-    PARQUET = "parquet"
-    MULTIJSON = "multijson"
-    ORC = "orc"
-    TSVE = "tsve"
-    RAW = "raw"
-    W3CLOGFILE = "w3clogfile"
+    CSV = ("csv", IngestionMappingType.CSV, False, True)
+    TSV = ("tsv", IngestionMappingType.CSV, False, True)
+    SCSV = ("scsv", IngestionMappingType.CSV, False, True)
+    SOHSV = ("sohsv", IngestionMappingType.CSV, False, True)
+    PSV = ("psv", IngestionMappingType.CSV, False, True)
+    TXT = ("txt", IngestionMappingType.UNKNOWN, False, True)
+    TSVE = ("tsve", IngestionMappingType.CSV, False, True)
+    JSON = ("json", IngestionMappingType.JSON, True, True)
+    SINGLEJSON = ("singlejson", IngestionMappingType.JSON, True, True)
+    MULTIJSON = ("multijson", IngestionMappingType.JSON, True, True)
+    AVRO = ("avro", IngestionMappingType.AVRO, True, True)
+    PARQUET = ("parquet", IngestionMappingType.PARQUET, False, False)
+    ORC = ("orc", IngestionMappingType.ORC, False, False)
+    RAW = ("raw", IngestionMappingType.UNKNOWN, False, True)
+    W3CLOGFILE = ("w3clogfile", IngestionMappingType.W3CLOGFILE, False, True)
+
+    def __init__(self, value_: str, ingestion_mapping_type: IngestionMappingType, mapping_required: bool, compressible: bool):
+        self.value_ = value_ # Formatted how Kusto Service expects it
```

Comment:
I also considered calling this `kusto_value` or `service_value` or `service_formatted_value` (in order of my preference).

Thoughts?

Reply:
I don't like value_, I like your suggestions (in the same order :) )